### PR TITLE
tiflash: update expression list supported by tiflash

### PR DESCRIPTION
**Pull request by ran-huang**
### What is changed, added or deleted? (Required)

Update the expression list supported by TiFlash.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [x] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

**If you select two or more versions from above**, to trigger the bot to cherry-pick this PR to your desired release version branch(es), you **must** add corresponding labels such as **needs-cherry-pick-4.0**, **needs-cherry-pick-3.1**, **needs-cherry-pick-3.0**, and **needs-cherry-pick-2.1**.

### What is the related PR or file link(s)?

- This PR is translated from:https://github.com/pingcap/docs-cn/pull/2921
- Other reference link(s):

### CODE REVIEW

File: tiflash/use-tiflash.md [hunk 1]
```diff
@@ -192,10 +192,17 @@ You can configure this parameter in either of the following ways:
...
 
+> **Note:**
+>
+> The feature that enables TiFlash to support the new framework for collations in TiDB is in development. Currently, if you enable the [new framework for collations](/character-set-and-collation.md#new-framework-for-collations) in TiDB, none of the expressions can be pushed down. This restriction will be removed in later versions.
+
 TiFlash mainly supports predicate and aggregate push-down calculations. Push-down calculations can help TiDB perform distributed acceleration. Currently, table joins and `DISTINCT COUNT` are not the supported calculation types, which will be optimized in later versions.
 
-Currently, TiFlash supports the limited push-down of common expressions. To learn the specific push-down expressions, refer to [expression list](https://github.com/pingcap/tidb/blob/release-3.1/expression/expression.go#L409).
+Currently, TiFlash supports the limited push-down of common expressions. To learn the specific push-down expressions, refer to [expression list](https://github.com/pingcap/tidb/blob/692e0098b1207ef26ea18bedfcc9ba067604da3c/expression/expression.go#L1115).
+
```
Comment (windtalker) [inline tiflash/use-tiflash.md:202]: the link is different from each version(master, 4.0, 3.1), the current link is for the master branch.

Comment (ran-huang) [reply tiflash/use-tiflash.md:202]: Noted. I'll edit the link when this PR is cherry-picked to other branches.

File: tiflash/use-tiflash.md [hunk 2]
```diff
@@ -192,10 +192,17 @@ You can configure this parameter in either of the following ways:
...
 ## Supported push-down calculations
 
+> **Note:**
+>
+> The feature that enables TiFlash to support the new framework for collations in TiDB is in development. Currently, if you enable the [new framework for collations](/character-set-and-collation.md#new-framework-for-collations) in TiDB, none of the expressions can be pushed down. This restriction will be removed in later versions.
+
 TiFlash mainly supports predicate and aggregate push-down calculations. Push-down calculations can help TiDB perform distributed acceleration. Currently, table joins and `DISTINCT COUNT` are not the supported calculation types, which will be optimized in later versions.
 
-Currently, TiFlash supports the limited push-down of common expressions. To learn the specific push-down expressions, refer to [expression list](https://github.com/pingcap/tidb/blob/release-3.1/expression/expression.go#L409).
+Currently, TiFlash supports the limited push-down of common expressions. To learn the specific push-down expressions, refer to [expression list](https://github.com/pingcap/tidb/blob/692e0098b1207ef26ea18bedfcc9ba067604da3c/expression/expression.go#L1115).
```
Comment (yikeke) [inline tiflash/use-tiflash.md:201]:
```suggestion
Currently, TiFlash supports pushing down a limited number of expressions. To learn the supported expressions, refer to [expression list](https://github.com/pingcap/tidb/blob/692e0098b1207ef26ea18bedfcc9ba067604da3c/expression/expression.go#L1115).
```

**Pull request merged**